flink-optimizer/src/main/java/org/apache/flink/optimizer/plantranslate/JobGraphGenerator.java

```diff
@@ -1141,9 +1141,6 @@ private DistributionPattern connectJobVertices(Channel channel, int inputNumber,
 			final DataDistribution dataDistribution = channel.getDataDistribution();
 			if (dataDistribution != null) {
 				sourceConfig.setOutputDataDistribution(dataDistribution, outputIndex);
-			} else {
```

Comment:
Can we set a special marker data distribution that indicates a sampled distribution?